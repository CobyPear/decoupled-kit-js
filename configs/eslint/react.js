/** @type {import('eslint').ESLint.ConfigData} */
module.exports = {
	plugins: ['react'],
	extends: [
		'./index.js',
		'plugin:react/recommended',
		'plugin:react-hooks/recommended',
		'plugin:react/jsx-runtime',
	],
	settings: {
		react: {
			version: 'detect',
		},
	},
};
