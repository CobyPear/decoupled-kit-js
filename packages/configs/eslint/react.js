module.exports = {
	plugins: ['react'],
	extends: [
		'./index.js',
		'plugin:react/recommended',
		'plugin:react-hooks/recommended',
	],
	settings: {
		react: {
			version: 'detect',
		},
	},
};
