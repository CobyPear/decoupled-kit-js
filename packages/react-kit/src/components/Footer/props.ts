export type FooterProps = Readonly<{
	/**
	 * A logo component to render in the footer.
	 * @example
	 * ```tsx
	 * export const Logo = ({ exampleLogo }: { exampleLogo: string }) => {
	 * 	// use the image component from your framework if applicable.
	 * 	return (
	 * 		<img src={exampleLogo} width="48" height="48" alt="Example Logo" />
	 * 	);
	 * };
	 * ```
	 *
	 */
	Logo:
		| { src: string; alt: string; href: string; styles?: string }
		| React.ReactElement
		| JSX.Element;
	/**
	 * Styles to be passed to the footer component.
	 */
	className?: string;
	/**
	 *  An instance of React.ReactChild
	 */
	children: JSX.Element;
}>;
