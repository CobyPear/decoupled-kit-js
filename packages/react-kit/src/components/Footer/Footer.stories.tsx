import PantheonLogo from '@assets/pantheon-fist-white.svg';
import { Footer as BottomSignature } from '@components/Footer';
import type { Meta, StoryObj } from '@storybook/react';
import { type FooterProps } from './props';

const Footer = ({ Logo, Content }: Pick<FooterProps, 'Logo' | 'Content'>) => {
	return (
		<div>
			<BottomSignature Logo={Logo} Content={Content} />
		</div>
	);
};

const meta: Meta<typeof Footer> = {
	title: 'Footer/Footer',
	component: Footer,
	parameters: {
		layout: 'fullscreen',
	},
	args: {
		Logo: { src: PantheonLogo, alt: 'Pantheon Systems', href: '/' },
		Content: {
			title: 'Pantheon Decoupled Kit',
			copy: 'Â© Pantheon 2023',
			builtWith: 'Pantheon.io',
			builtWithLink: 'https://pantheon.io/',
		},
	},
	argTypes: {},
	tags: ['autodocs'],
};

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {};
