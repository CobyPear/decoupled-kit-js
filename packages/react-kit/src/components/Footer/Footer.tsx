import clsx from 'clsx';
import React from 'react';
import { FooterProps } from './props';

/**
 * @see {@link https://live-storybook-react-kit.appa.pantheon.site/?path=/docs/footer-footer--docs}
 */
export const Footer = ({ Logo, className, children }: FooterProps) => {
	const FooterLogo = () => {
		if (React.isValidElement(Logo)) {
			return Logo;
		} else if (typeof Logo === 'object' && 'src' in Logo && 'alt' in Logo) {
			const { src, alt } = Logo;
			return (
				<div
					className={Logo?.styles || 'rk-mx-auto rk-block rk-pb-3.5 rk-pt-14'}
				>
					<img src={src} alt={alt} className={'rk-inline rk-h-16 rk-w-16'} />
				</div>
			);
		}
		return null;
	};
	return (
		<footer
			className={clsx(
				className,
				'rk-bg-neutral-800 rk-text-center rk-text-white',
			)}
		>
			<FooterLogo />
			{children}
		</footer>
	);
};
