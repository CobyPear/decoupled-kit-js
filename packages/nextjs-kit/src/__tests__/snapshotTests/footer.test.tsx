import { render } from '@testing-library/react';

import Footer from '../../components/footer';

/**
 * @vitest-environment jsdom
 */

const footerMenu = [
  {
    label: 'Home',
    path: '/',
    id: 'homeTest',
  },
  {
    label: 'Posts',
    path: '/posts',
    id: 'postsTest',
  },
  {
    label: 'HPagesome',
    path: '/pages',
    id: 'homePages',
  },
];

describe('<Footer />', () => {
  it("should render 'footer'", () => {
    const { asFragment } = render(
      <Footer menuItems={footerMenu}>
        <span className="mx-auto">
          Â© {new Date().getFullYear()} Built with{' '}
          <a
            className="text-white hover:text-blue-100 underline"
            href="https://nextjs.org/"
          >
            Next.js
          </a>{' '}
          and{' '}
          <a
            className="text-blue-500 underline hover:text-blue-100"
            href="https://www.wordpress.com/"
          >
            Wordpress
          </a>
        </span>
      </Footer>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
