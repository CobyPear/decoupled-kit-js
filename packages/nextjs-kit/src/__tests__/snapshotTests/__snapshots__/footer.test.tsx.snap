// Vitest Snapshot v1

exports[`<Footer /> > should render 'footer' 1`] = `
<DocumentFragment>
  <footer
    class="w-full text-white bg-black p-4 mt-12"
  >
    <nav
      class="flex flex-col max-w-lg mx-auto lg:max-w-screen-lg"
    >
      <ul>
        <ul>
          <li
            class="list-disc text-blue-300"
          >
            <a
              class="text-blue-300 hover:underline hover:text-blue-100 focus:text-purple-600 active:text-purple-300"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="list-disc text-blue-300 ml-3"
          >
            <a
              class="text-blue-300 hover:underline hover:text-blue-100 focus:text-purple-600 active:text-purple-300"
              href="/articles"
            >
              Articles
            </a>
          </li>
        </ul>
        <li
          class="list-disc text-blue-300"
        >
          <a
            class="text-blue-300 hover:underline hover:text-blue-100 focus:text-purple-600 active:text-purple-300"
            href="/pages"
          >
            Pages
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="flex my-4 p-2"
    >
      <span
        class="mx-auto"
      >
        Â© 2022 Built with 
        <a
          class="text-white hover:text-blue-100 underline"
          href="https://nextjs.org/"
        >
          Next.js
        </a>
         and 
        <a
          class="text-blue-500 underline hover:text-blue-100"
          href="https://www.wordpress.com/"
        >
          Wordpress
        </a>
      </span>
    </div>
  </footer>
</DocumentFragment>
`;
